iroh/src/node.rs

```diff
@@ -299,6 +305,41 @@ impl<D: iroh_blobs::store::Store> NodeInner<D> {
             });
         }
 
+        if let Some(nodes_data_path) = nodes_data_path {
+            let ep = self.endpoint.clone();
+            let token = self.cancel_token.clone();
+
+            join_set.spawn(async move {
+                let mut save_timer = tokio::time::interval_at(
+                    tokio::time::Instant::now() + SAVE_NODES_INTERVAL,
+                    SAVE_NODES_INTERVAL,
+                );
+
+                loop {
+                    tokio::select! {
+                        biased;
+                        _ = token.cancelled() => {
+                            trace!("save known node addresses shutdown");
+                            let addrs = node_addresses_for_storage(&ep);
+                            if let Err(err) = store_node_addrs(&nodes_data_path, &addrs).await {
+                                warn!("failed to store knonw node addresses: {:?}", err);
+                            }
+                            break;
+                        }
+                        _ = save_timer.tick() => {
+                            trace!("save known node addresses tick");
+                            let addrs = node_addresses_for_storage(&ep);
+                            if let Err(err) = store_node_addrs(&nodes_data_path, &addrs).await {
+                                warn!("failed to store knonw node addresses: {:?}", err);
+                            }
+                        }
+                    }
+                }
```

Comment:
Please always attach a tracing span when spawning a new tokio task.  Otherwise all the span information is gone.  If a new span is not suitable than `.instrument(Span::current())` propagates the current span, but here a new span makes sense: `.instrument(info_span!("known-addrs"))` or so.